inc/featured-content.php

```diff
@@ -116,3 +116,69 @@ function largo_have_homepage_featured_posts() {
 	return $featured_query->have_posts();
 
 }
+
+/**
+ * Get posts marked as "Featured in category" for a given category name.
+ *
+ * @param string $category_name the category to retrieve featured posts for.
+ * @param integer $number total number of posts to return, backfilling with regular posts as necessary.
+ * @since 0.5
+ */
+function largo_get_featured_posts_in_category( $category_name, $number = 5 ) {
+	$args = array(
+		'category_name' => $category_name,
+		'numberposts' => $number,
+		'post_status' => 'publish',
+	);
+
+	$tax_query = array(
+		'tax_query' => array(
+			array(
+				'taxonomy' => 'prominence',
+				'field' => 'slug',
+				'terms' => 'category-featured',
+			)
+		)
+	);
+
+	// Get the featured posts
+	$featured_posts = get_posts( array_merge( $args, $tax_query ) );
+
+	// Backfill with regular posts if necessary
+	if ( count( $featured_posts ) < (int) $number ) {
+		$needed = (int) $number - count( $featured_posts );
+		$regular_posts = get_posts( array_merge( $args, array(
+			'numberposts' => $needed,
+			'post__not_in' => array_map( function( $x ) { return $x->ID; }, $featured_posts )
+		)));
+		$featured_posts = array_merge( $featured_posts, $regular_posts );
+	}
+
+	return $featured_posts;
+}
+
+/**
+ * Helper for getting posts in a category archive, excluding featured posts.
+ * 
+ * @param WP_Query $query
+ * @uses largo_get_featured_posts_in_category
+ * @since 0.4
+ */
+function largo_category_archive_posts( $query ) {
+	// don't muck with admin, non-categories, etc
+	if ( ! $query->is_category() || ! $query->is_main_query() || is_admin() ) return;
+
+	// If this has been disabled by an option, do nothing
+	if ( of_get_option( 'hide_category_featured' ) == true ) return;
+
+	// get the featured posts
+	$featured_posts = largo_get_featured_posts_in_category( $query->get( 'category_name' ) );
+
+	// get the IDs from the featured posts
+	$featured_post_ids = array();
+	foreach ( $featured_posts as $fpost )
+		$featured_post_ids[] = $fpost->ID;
+
+	$query->set( 'post__not_in', $featured_post_ids );
```

Comment:
Note that this stomps any existing `post_not_in` IDs, but that's okay, because no standard query should have set those. If those IDs were set by a filter at some previous point in this query, then this part of the function won't run because `is_main_query()` will have returned false, because the query was modified.